### PR TITLE
Improved message when web vitals not available

### PR DESCRIPTION
**Pull request by tunetheweb**
We occasionally see reviews, or complaints on Twitter like this:

image

I noticed on our Corp machines when looking at some sites (e.g. www.youtube.com) we get a `This page cannot be scripted due to an ExtensionsSettings policy.` error in the service worker console, so suspect this is the cause of a lot of these errors.

~I think a quick fix here, rather than surfacing the exact error and feeding it through from the service worker to the popup page~ Actually I changed my mind and did it the full way after all.

image

Fixes https://github.com/GoogleChrome/web-vitals-extension/issues/118
Fixes https://github.com/GoogleChrome/web-vitals-extension/issues/54
Fixes https://github.com/GoogleChrome/web-vitals-extension/issues/56
As handles exceptions better and so switched from `console.error` to `console.log` as no need to surface them in the `chrome://extensions` page anymore.

### CODE REVIEW

File: service_worker.js [hunk 1]
```diff
@@ -49,6 +54,11 @@ function getWebVitals(tabId) {
   }, (result) => {
     // Catch errors such as "This page cannot be scripted due
     // to an ExtensionsSettings policy."
+    let error = chrome.runtime.lastError;
+    if (error && error.message) {
+      console.log(error.message);
+      chrome.tabs.get(tabId, (tab) => setExtensionErrorMessage(tab, error.message));
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply service_worker.js:60]: This is using the `tabid` to get the `tab`.

File: service_worker.js [hunk 2]
```diff
@@ -49,6 +54,11 @@ function getWebVitals(tabId) {
   }, (result) => {
     // Catch errors such as "This page cannot be scripted due
     // to an ExtensionsSettings policy."
+    let error = chrome.runtime.lastError;
+    if (error && error.message) {
+      console.log(error.message);
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply service_worker.js:59]: Interesting question. They are two different consoles and I find it annoying to have to look at both. In reality most people won't be looking at these consoles unless of errors so I'd rather have more logging.

File: service_worker.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (brendankenny) [reply service_worker.js:60]: Missed that, ignore :)

File: src/browser_action/chrome.js [hunk 3]
```diff
@@ -30,10 +30,15 @@ export function loadLocalMetrics() {
 
         chrome.storage.local.get(key, result => {
           if (result[key] !== undefined) {
-            resolve({
-              metrics: result[key],
-              background: tabLoadedInBackground
-            });
+            if (typeof(result[key]) === 'string') {
+              // It's an error message, not a metrics object
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply src/browser_action/chrome.js]: Fixed in https://github.com/GoogleChrome/web-vitals-extension/pull/144/commits/0d7d0a14447aebe7b7b15917d400f80227199472.

I spent ages trying to do this and just couldn't get it to work. But worked first time here. Sometimes you need to just take a break!!

**Pull request merged**